pkg/kf/internal/genericcli/xargs.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genericcli
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"os/exec"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/google/kf/v2/pkg/kf/commands/config"
+	"github.com/google/kf/v2/pkg/kf/spaces"
+	"github.com/segmentio/textio"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"knative.dev/pkg/injection/clients/dynamicclient"
+	"knative.dev/pkg/logging"
+)
+
+// NewListCommand creates a list command that can print tables.
+func NewXargsCommand(t Type, p *config.KfParams, spacesClient spaces.Client, opts ...XargsOption) *cobra.Command {
+	options := XargsOptions{
+		WithXargsPluralFriendlyName(t.FriendlyName() + "s"),
+		WithXargsCommandName(fmt.Sprintf("xargs-%s", strings.ToLower(t.FriendlyName()+"s"))),
+	}.Extend(opts)
+
+	flags := &XargFlags{}
+
+	labelSelectorFlags := map[string]*string{}
+
+	var argNames string
+	var requiredArgs int
+	var maxArgs int
+	argFilters := options.ArgumentFilters()
+	for i := len(argFilters) - 1; i >= 0; i-- {
+		argNames = strings.TrimSpace(fmt.Sprintf("%s %s", argFilters[i].Name, argNames))
+		maxArgs++
+		if argFilters[i].Required {
+			argNames = fmt.Sprintf("%s", argNames)
+			requiredArgs++
+		} else {
+			argNames = fmt.Sprintf("[%s]", argNames)
+		}
+	}
+
+	var example string
+	if options.Example() != "" {
+		example = options.Example()
+	} else {
+		example = fmt.Sprintf("kf %s", options.CommandName())
+	}
+
+	var short string
+	if options.Short() != "" {
+		short = options.Short()
+	} else {
+		short = fmt.Sprintf("Xarg %s.", options.PluralFriendlyName())
+	}
+
+	var long string
+	if options.Long() != "" {
+		long = options.Long()
+	} else {
+		long = fmt.Sprintf("Runs a command on each %s in targeted spaces.", t.FriendlyName())
```

Comment:
I think this needs two variants, one for if the resource is cluster scoped (not in a space) and one for if it's not.

Reply:
Done.